File: src/ch1/coin.py
# 어느 가계의 계산 카운터에 500원짜리 10개, 100원짜리 3개, 50원짜리 10개 있다. 잔돈으로 3950원을 거슬러 줘야 할 경우 나올 수 있는 모든 조합을 나열하시요.

# 거스름돈
price = 3950

for i500 in range(0, 11):
  for i100 in range(0, 4):
    for i50 in range(0, 11):
      if (i500 * 500 + i100 * 100 + i50 * 50) == price:
        print(f"500원 {i500}개, 100원 {i100}개, 50원 {i50}개")

File: src/ch1/coin.rs
// 러스트로 거스름돈 조합 계산
fn main() {
  // 거스름돈
  let price = 3950;
  for i500 in 0..11 {
    for i100 in 0..4 {
      for i50 in 0..11 {
        let total = i500 * 500 + i100 * 100 + i50 * 50;
        if total == price {
          println!("500원 {}개, 100원 {}개, 50원 {}개", i500, i100, i50);
        }
      }
    }
  }
}